pkg/store/proxy.go

```diff
@@ -657,14 +561,118 @@ func (s *ProxyStore) LabelValues(ctx context.Context, r *storepb.LabelValuesRequ
 			return nil
 		})
 	}
+	level.Debug(reqLogger).Log("msg", "LabelValues: started fanout streams", "status", strings.Join(storeDebugMsgs, ";"))
 
 	if err := g.Wait(); err != nil {
 		return nil, err
 	}
 
-	level.Debug(s.logger).Log("msg", strings.Join(storeDebugMsgs, ";"))
 	return &storepb.LabelValuesResponse{
 		Values:   strutil.MergeUnsortedSlices(all...),
 		Warnings: warnings,
 	}, nil
 }
+
+func storeInfo(st Client) (storeID string, storeAddr string, isLocalStore bool) {
```

Comment:
Why not make this a struct function?

```suggestion
func (st *Client) Info() (storeID string, storeAddr string, isLocalStore bool) {
```

Also, I'm not a big fan of the use of multiple named returns like this. For example, I think it would be better to implement individual funcs like `func (st *Client) IsLocalStore() bool {...}`.